src/main/java/jpa/dao/CompanyDao.java

```diff
@@ -0,0 +1,75 @@
+package jpa.dao;
+
+import jpa.entity.Companies;
+import lombok.extern.log4j.Log4j;
+
+import javax.persistence.EntityManager;
+
+@Log4j
+public class CompanyDao {
```

Comment:
Please use interface for the contract. And this class should be the implementation of the contract.